backend/metrics_backend/metrics_service.py

```diff
@@ -203,17 +204,33 @@ def handle_token_network_created(self, event):
 
         if not self.follows_token_network(token_network_address):
             log.info(f'Found token network for token {token_address} @ {token_network_address}')
-            self.create_token_network_for_address(token_network_address, event_block_number)
+            self.create_token_network_for_address(
+                token_network_address,
+                token_address,
+                event_block_number,
+            )
 
     def create_token_network_for_address(
         self,
         token_network_address: Address,
+        token_address: Address,
         block_number: int = 0,
     ):
-        token_network = TokenNetwork(token_network_address)
+        # get token infos
+        token_proxy = self.web3.eth.contract(
+            address=token_address,
+            abi=CONTRACT_MANAGER.get_contract_abi(CONTRACT_HUMAN_STANDARD_TOKEN),
+        )
+        token_infos = TokenInfo(
+            token_proxy.functions.name().call(),
```

Comment:
I don't know how exactly call works but name, symbol, and decimals are optional properties and might be missing from a token network, I am guessing that the call with throw, is that right?

If missing we should return an empty string  on name and symbol.

For decimals the default if missing should be 18. https://github.com/raiden-network/raiden/issues/2197#issuecomment-415718934

Reply:
Good point, I updated the PR to handle these cases.